src/ngx-smart-modal/src/components/ngx-smart-modal.component.ts

```diff
@@ -92,6 +99,79 @@ export class NgxSmartModalComponent implements OnInit, OnDestroy {
     this._sendEvent('delete');
   }
 
+  /**
+  * Set positionX and positionY to save last position of dragged modal
+  */
+  public setPosition(posX: number, posY: number) {
+    this.positionX = posX;
+    this.positionY = posY;
+  }
+
+  /**
+  * Moves dialog by changing top and left style of modal dialog by offset
+  */
+  public moveDialog(offsetX: number, offsetY: number) {
+    if (!this.nsmDialog.length) {
+      return false;
+    }
+    this.nsmDialog.last.nativeElement.style.top = (this.nsmDialog.last.nativeElement.offsetTop - offsetY) + "px";
+    this.nsmDialog.last.nativeElement.style.left = (this.nsmDialog.last.nativeElement.offsetLeft - offsetX) + "px";
+    return true;
+  }
+
+  /**
+  * Listens for mouse down event to initiate dragging of the modal
+  */
+  @HostListener('document:mousedown', ['$event'])
+  public startDrag(e: MouseEvent) {
```

Comment:
For me, there is no reason to be public

Reply:
Tests are written for public methods.

Reply:
Yes, and after reflexion, those functions have to be private

Reply:
Okay. I will change my tests and make them private. Your hostListener (targetPlacement) is also public btw.